sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -0,0 +1,407 @@
+            {
+                sb.EntityPath = eventHubName;
+            }
+
+            var client = new EventHubProducerClient(sb.ToString());
+            AddEventHubProducerClient(eventHubName, client);
+        }
+
+        /// <summary>
+        /// Add a connection for listening on events from an event hub. Connect via the connection string and use the SDK's built-in storage account.
+        /// </summary>
+        /// <param name="eventHubName">name of the event hub</param>
+        /// <param name="receiverConnectionString">connection string for receiving messages. This can encapsulate other service bus properties like the namespace and endpoints.</param>
+        public void AddReceiver(string eventHubName, string receiverConnectionString)
+        {
+            if (eventHubName == null)
+            {
+                throw new ArgumentNullException(nameof(eventHubName));
+            }
+            if (receiverConnectionString == null)
+            {
+                throw new ArgumentNullException(nameof(receiverConnectionString));
+            }
+
+            this._receiverCreds[eventHubName] = new ReceiverCreds
+            {
+                EventHubConnectionString = receiverConnectionString
+            };
+        }
+
+        /// <summary>
+        /// Add a connection for listening on events from an event hub. Connect via the connection string and use the supplied storage account
+        /// </summary>
+        /// <param name="eventHubName">name of the event hub</param>
+        /// <param name="receiverConnectionString">connection string for receiving messages</param>
+        /// <param name="storageConnectionString">storage connection string that the EventProcessorHost client will use to coordinate multiple listener instances. </param>
+        public void AddReceiver(string eventHubName, string receiverConnectionString, string storageConnectionString)
+        {
+            if (eventHubName == null)
+            {
+                throw new ArgumentNullException(nameof(eventHubName));
+            }
+            if (receiverConnectionString == null)
+            {
+                throw new ArgumentNullException(nameof(receiverConnectionString));
+            }
+            if (storageConnectionString == null)
+            {
+                throw new ArgumentNullException(nameof(storageConnectionString));
+            }
+
+            this._receiverCreds[eventHubName] = new ReceiverCreds
+            {
+                EventHubConnectionString = receiverConnectionString,
+                StorageConnectionString = storageConnectionString
+            };
+        }
+
+        internal EventHubProducerClient GetEventHubProducerClient(string eventHubName, string connection)
+        {
+            EventHubProducerClient client;
+
+            if (string.IsNullOrEmpty(eventHubName))
+            {
+                EventHubsConnectionStringBuilder builder = new EventHubsConnectionStringBuilder(connection);
```

Comment:
I'd recommend using the `EventHubConnectionStringProperties` type here;  it'll be released with November's beta.

Reply:
You also may want to consider validating, as it is pretty rare for customers to find the connection string with the Event Hub name embedded.    Not sure if this is already validated elsewhere, but I suspect this could end up with a missing Event Hub name.